src/handwritten/_grpc_time.py

```diff
@@ -27,24 +52,21 @@ def convert_time_to_timestamp(dt: Union[std_datetime, ht_datetime], ts: GrpcTime
         if remainder_yoctoseconds >= _YS_PER_NS / 2:
             nanos += 1
     else:
-        nanos = utc_dt.microsecond * _NS_PER_US
+        nanos = dt.microsecond * _NS_PER_US
 
-    ts.FromNanoseconds(seconds * _NS_PER_S + nanos)
+    ts.FromNanoseconds(seconds_since_1970 * _NS_PER_S + nanos)
+    return ts
 
 
+def timestamp_to_1970_epoch(timestamp, yoctoseconds):
```

Comment:
If you extract the epoch into a constant, this becomes a one-liner, so consider inlining it into `convert_timestamp_to_time`.

If not: `_` prefix, type hints